app/main.py

```diff
@@ -1,8 +1,55 @@
 class Car:
-    # write your code here
-    pass
+
+    def __init__(
+            self,
+            comfort_class: int,
+            clean_mark: int,
+            brand: str
+    ) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+
+    def __init__(
+            self,
+            distance_from_city_center: float,
+            clean_power: int,
+            average_rating: float,
+            count_of_ratings: int
+    ) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list[Car]) -> int:
+        final_price = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                final_price += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+        return final_price
+
+    def calculate_washing_price(self, car: Car) -> float:
+        return round(
+            car.comfort_class * (
+                self.clean_power - car.clean_mark
+            ) * (
+                self.average_rating / self.distance_from_city_center
+            ), 1
+        )
```

Comment:
If you have some long math, you can split it onto additional variables, or break after binary operations (not before - it cause the W504 errors) read checklist

Reply:
solved